packages/dd-trace/src/encode/0.4.js

```diff
@@ -141,8 +141,25 @@ function getFields () {
   }, {})
 }
 
+function cachedString (str) {
+  if (stringCache[str]) {
+    return stringCache[str]
+  }
+
+  const buf = Buffer.from(str, 'utf-8')
+  const token = tokens.getStringPrefix(buf.length)
+  const prefixed = Buffer.allocUnsafe(buf.length + token.length)
```

Comment:
Could the size of the buffer be calculated from the string alone to avoid creating a buffer just for the string and then this other buffer to copy the string buffer to it?

Reply:
Yes. Done that in latest push.